internal/policy/policy.go

```diff
@@ -230,6 +260,50 @@ func (s *State) FindAuthorizedSigningKeyIDs(ctx context.Context, roleName string
 	return entry.KeyIDs, nil
 }
 
+func (s *State) FindPublicKeysForPath(ctx context.Context, path string) ([]*tuf.Key, error) {
+	if err := s.Verify(ctx); err != nil {
+		return nil, err
+	}
+
+	targetsMetadata, err := s.GetTargetsMetadata(TargetsRoleName)
+	if err != nil {
+		return nil, err
+	}
+
+	allPublicKeys := targetsMetadata.Delegations.Keys
+	delegationsQueue := targetsMetadata.Delegations.Roles
+
+	trustedKeys := []*tuf.Key{}
+	for {
+		if len(delegationsQueue) <= 1 {
+			return trustedKeys, nil
+		}
+
+		delegation := delegationsQueue[0]
+		delegationsQueue = delegationsQueue[1:]
+
+		if delegation.Matches(path) {
+			for _, keyID := range delegation.KeyIDs {
+				key := allPublicKeys[keyID]
+				trustedKeys = append(trustedKeys, key)
+			}
+
+			// TODO: clarify logic for delegation.Terminating
+
+			if s.HasTargetsRole(delegation.Name) {
+				delegatedMetadata, err := s.GetTargetsMetadata(delegation.Name)
+				if err != nil {
+					return nil, err
+				}
+				for keyID, key := range delegatedMetadata.Delegations.Keys {
+					allPublicKeys[keyID] = key
```

Comment:
Might be worth adding more comments explaining why we're doing this (it looks like this is some sort of override?)

Is it a problem if this key id has already been pulled out into trustedkeys from a previous iteration of the delegation queue? (is this even possible?)

Reply:
(Addressing outstanding comments for https://github.com/adityasaky/gittuf/pull/62.)

I'm tracking all the public keys encountered so that when we're on a delegation matched path, we can look up the keys from the global map to populate trusted keys. This is because gittuf differs from regular TUF where all keys along a matched delegation path are trusted rather than just the keys listed in the leaf delegation. The overwriting shouldn't impact the _value_ of the key which matters. trustedKeys may end up with duplicates but that doesn't affect threshold verification luckily. That said, I can swap that out to also be a map[key]bool to ape set behavior.